Update hardware-and-software-requirements-for-biztalk-server-2020.md (#1006)
MicrosoftDocs/biztalk-docs: biztalk/core/sftp-adapter.md

title: "SFTP Adapter"
description: Create or configure a receive location and send port using the SFTP adapter in BizTalk Server, including FAQs using the SFTP adapter
ms.custom: "biztalk-2020"
ms.date: "02/11/2025"
ms.service: biztalk-server
ms.reviewer: "niklase"
ms.suite: ""
ms.topic: "article"
---
# SFTP Adapter
BizTalk Server includes an **SFTP** adapter to send and receive messages from a secure FTP server using the SSH file transfer protocol. This topic includes the steps to configure an **SFTP** receive location, and configure an SFTP send port to receive and send messages from a secure FTP server. It also includes common questions and answers.

## Prerequisites

> **Starting with BizTalk Server 2016**, the SFTP adapter uses WinSCP to connect to SFTP, and therefore supports a larger range of SFTP servers. **Download [WinSCP](http://winscp.net)** automation.zip file,under .NET assembly COM library section or [older downloads](https://sourceforge.net/projects/winscp/files/WinSCP/) , on the BizTalk Server runtime and copy the content to the BizTalk installation folder. Be sure to check the supported WinSCP versions for each BizTalk Server version:

Reviewer: - Isn't this a duplicate change from https://github.com/MicrosoftDocs/biztalk-docs/pull/1005?
- Use bold only for UX elements, inline code elements, or file names:
- I went to http://winscp.net, but couldn't find the ".NET assembly COM library section". The only download link is for a **WinSCP-6.3.6-Setup.exe** file, not an **automation.zip** file?

```suggestion

Starting with BizTalk Server 2016, the SFTP adapter connects to SFTP by using WinSCP, which supports a larger range of SFTP servers.

1. Visit the [WinSCP website](http://winscp.net) or the [SourceForge downloads website](https://sourceforge.net/projects/winscp/files/WinSCP/), and download the **automation.zip** file onto the BizTalk Server runtime.

1. Copy the content to the BizTalk installation folder. For the BizTalk Server version that you use, make sure that you check the default supported WinSCP version:

```

Author: - It is not really duplicate, but one section has this file version so perhaps best to only have this change since it is later? 
- Bold : I have not done changes to bold. This is since before same so not doing any changes on that. But we can change that. But perhaps we do that later separately to focus on main change here. 
- There is automation/COM library in other download section. I have linked to good exact locations in SFTP adapter for each version. So I think the "under .NET assembly COM library section" is very important that you suggested to omit here. We must have this.

Author: I have changed to this which is more like your version, but with more details and the steps in correct order. 

Starting with BizTalk Server 2016, the SFTP adapter connects to SFTP by using WinSCP library, which supports a larger range of SFTP servers.
1. Visit the [WinSCP website](https://winscp.net/eng/downloads.php), under .NET assembly COM library section, or the [SourceForge downloads website](https://sourceforge.net/projects/winscp/files/WinSCP/) for older versions. 
2. Be sure to use the supported WinSCP versions for each BizTalk Server version:
 * BizTalk Server 2020 CU6 - WinSCP version [6.3.5](https://sourceforge.net/projects/winscp/files/WinSCP/6.3.5/WinSCP-6.3.5-Automation.zip/download)
 * BizTalk Server 2020 CU5 - WinSCP version [6.1.2](https://sourceforge.net/projects/winscp/files/WinSCP/6.1.2/WinSCP-6.1.2-Automation.zip/download)
 * BizTalk Server 2020 CU3 or CU4 - WinSCP version [5.19.2](https://sourceforge.net/projects/winscp/files/WinSCP/5.19.2/WinSCP-5.19.2-Automation.zip/download)
 * BizTalk Server 2020 CU1 or CU2 - WinSCP version [5.17.6](https://sourceforge.net/projects/winscp/files/WinSCP/5.17.6/WinSCP-5.17.6-Automation.zip/download)
 * BizTalk Server 2020 RTM - WinSCP version [5.15.4](https://sourceforge.net/projects/winscp/files/WinSCP/5.15.4/WinSCP-5.15.4-Automation.zip/download)
 * BizTalk Server 2016 CU9 - WinSCP version [5.19.2](https://sourceforge.net/projects/winscp/files/WinSCP/5.19.2/WinSCP-5.19.2-Automation.zip/download)
 * BizTalk Server 2016 CU7 or CU8 - WinSCP version [5.15.9](https://sourceforge.net/projects/winscp/files/WinSCP/5.15.9/WinSCP-5.15.9-Automation.zip/download)
 * BizTalk Server 2016 CU5 or CU6 - WinSCP version [5.13.1](https://sourceforge.net/projects/winscp/files/WinSCP/5.13.1/WinSCP-5.13.1-Automation.zip/download)
 * BizTalk Server 2016 RTM to CU4 - WinSCP version [5.7.7](https://sourceforge.net/projects/winscp/files/WinSCP/5.7.7/WinSCP-5.7.7-Automation.zip/download)
3. Download the correct **WinSCP-version-Automation.zip** file onto the BizTalk runtime servers.
4. Decompress the **WinSCP-version-Automation.zip** contents, WinSCP.exe and WinSCPnet.dll, to the BizTalk installation folder. (%BTSINSTALLPATH%)

title: "SFTP Adapter"
description: Create or configure a receive location and send port using the SFTP adapter in BizTalk Server, including FAQs using the SFTP adapter
ms.custom: "biztalk-2020"
ms.date: "02/14/2025"
ms.service: biztalk-server
ms.reviewer: "niklase"
ms.suite: ""
ms.topic: "article"
---
# SFTP Adapter
BizTalk Server includes an **SFTP** adapter to send and receive messages from a secure FTP server using the SSH file transfer protocol. This topic includes the steps to configure an **SFTP** receive location, and configure an SFTP send port to receive and send messages from a secure FTP server. It also includes common questions and answers.

## Prerequisites

> Starting with BizTalk Server 2016, the SFTP adapter connects to SFTP by using WinSCP library, which supports a larger range of SFTP servers.

Reviewer: ```suggestion
Starting with BizTalk Server 2016, the SFTP adapter connects to SFTP by using the WinSCP library, which supports a larger range of SFTP servers.
```

> 1. Visit the [WinSCP website](https://winscp.net/eng/downloads.php), under .NET assembly COM library section, or the [SourceForge downloads website](https://sourceforge.net/projects/winscp/files/WinSCP/) for older versions. 

Reviewer: The URL goes to a page that doesn't have a .NET assembly COM library section.

Author: The zip file is downloaded fine for me in both locations. All customers find this fine and they have never complained about this so I don't think this is an issue.

> 1. Visit the [WinSCP website](https://winscp.net/eng/downloads.php), under .NET assembly COM library section, or the [SourceForge downloads website](https://sourceforge.net/projects/winscp/files/WinSCP/) for older versions. 

Reviewer: I see you have to scroll down further to find the section. I suggest bolding the section name to clarify what you're looking for:

```suggestion
1. Visit the [WinSCP website](https://winscp.net/eng/downloads.php) and find the **.NET assembly / COM library** section, or visit the [SourceForge downloads website](https://sourceforge.net/projects/winscp/files/WinSCP/) for older versions. 
```

> 2. Be sure to use the supported WinSCP versions for each BizTalk Server version:

Reviewer: Replace manual numbering with autonumbering syntax:

```suggestion
1. Make sure to use the supported WinSCP versions for each BizTalk Server version:
```

 * BizTalk Server 2020 CU3 or CU4 - WinSCP version [5.19.2](https://sourceforge.net/projects/winscp/files/WinSCP/5.19.2/WinSCP-5.19.2-Automation.zip/download)
 * BizTalk Server 2020 CU1 or CU2 - WinSCP version [5.17.6](https://sourceforge.net/projects/winscp/files/WinSCP/5.17.6/WinSCP-5.17.6-Automation.zip/download)
 * BizTalk Server 2020 RTM - WinSCP version [5.15.4](https://sourceforge.net/projects/winscp/files/WinSCP/5.15.4/WinSCP-5.15.4-Automation.zip/download)
 * BizTalk Server 2016 CU9 - WinSCP version [5.19.2](https://sourceforge.net/projects/winscp/files/WinSCP/5.19.2/WinSCP-5.19.2-Automation.zip/download)
 * BizTalk Server 2016 CU7 or CU8 - WinSCP version [5.15.9](https://sourceforge.net/projects/winscp/files/WinSCP/5.15.9/WinSCP-5.15.9-Automation.zip/download)
 * BizTalk Server 2016 CU5 or CU6 - WinSCP version [5.13.1](https://sourceforge.net/projects/winscp/files/WinSCP/5.13.1/WinSCP-5.13.1-Automation.zip/download)
 * BizTalk Server 2016 RTM to CU4 - WinSCP version [5.7.7](https://sourceforge.net/projects/winscp/files/WinSCP/5.7.7/WinSCP-5.7.7-Automation.zip/download)

> 3. Download the correct **WinSCP-version-Automation.zip** file onto the BizTalk runtime servers.

Reviewer: I didn't find any ZIP file.

```suggestion
1. Download the correct **WinSCP-version-Automation.zip** file onto the BizTalk runtime servers.
```

> 4. Decompress the **WinSCP-version-Automation.zip** contents, WinSCP.exe and WinSCPnet.dll, to the BizTalk installation folder. (%BTSINSTALLPATH%)

Reviewer: ```suggestion
1. Extract the **WinSCP-version-Automation.zip** contents, which contains WinSCP.exe and WinSCPnet.dll, to the BizTalk installation folder. (%BTSINSTALLPATH%)
```

> See also 

Reviewer: ```suggestion
For more information, see the following documentation:

```

> [Hardware and Software Requirements for BizTalk 2020](../install-and-config-guides/hardware-and-software-requirements-for-biztalk-server-2020.md)

Reviewer: ```suggestion
- [Hardware and Software Requirements for BizTalk 2020](../install-and-config-guides/hardware-and-software-requirements-for-biztalk-server-2020.md)
```

> [Hardware and Software Requirements for BizTalk 2016](../install-and-config-guides/hardware-and-software-requirements-for-biztalk-server-2016.md)

Reviewer: ```suggestion
- [Hardware and Software Requirements for BizTalk 2016](../install-and-config-guides/hardware-and-software-requirements-for-biztalk-server-2016.md)
```

> BizTalk Server 2013 and BizTalk Server 2013 R2 use older internal ssh library instead of WinSCP with limited SFTP server compatibility.

Reviewer: ```suggestion
> [!NOTE]
>
> BizTalk Server 2013 and BizTalk Server 2013 R2 use the older internal SSH library, which has limited SFTP server compatibility compared to WinSCP.
```

 * BizTalk Server 2020 CU3 or CU4 - WinSCP version [5.19.2](https://sourceforge.net/projects/winscp/files/WinSCP/5.19.2/WinSCP-5.19.2-Automation.zip/download)
 * BizTalk Server 2020 CU1 or CU2 - WinSCP version [5.17.6](https://sourceforge.net/projects/winscp/files/WinSCP/5.17.6/WinSCP-5.17.6-Automation.zip/download)
 * BizTalk Server 2020 RTM - WinSCP version [5.15.4](https://sourceforge.net/projects/winscp/files/WinSCP/5.15.4/WinSCP-5.15.4-Automation.zip/download)
 * BizTalk Server 2016 CU9 - WinSCP version [5.19.2](https://sourceforge.net/projects/winscp/files/WinSCP/5.19.2/WinSCP-5.19.2-Automation.zip/download)
 * BizTalk Server 2016 CU7 or CU8 - WinSCP version [5.15.9](https://sourceforge.net/projects/winscp/files/WinSCP/5.15.9/WinSCP-5.15.9-Automation.zip/download)
 * BizTalk Server 2016 CU5 or CU6 - WinSCP version [5.13.1](https://sourceforge.net/projects/winscp/files/WinSCP/5.13.1/WinSCP-5.13.1-Automation.zip/download)
 * BizTalk Server 2016 RTM to CU4 - WinSCP version [5.7.7](https://sourceforge.net/projects/winscp/files/WinSCP/5.7.7/WinSCP-5.7.7-Automation.zip/download)

> or in [Hardware and Software Requirements](../install-and-config-guides/hardware-and-software-requirements-for-biztalk-server-2020.md)

Reviewer 2: If you choose to accept my other suggestion to not list the WinSCP version on the Hardware and Software Requirements page, then this last statement can be removed.

Author: Done. Referring to SFTP-Adapter documentation in one place

    | User name | Specifies a username for the secure FTP server.|

    **SSH Server**

    | Use this | To do this|
    |--------------|----------------|
    | Append If Exists | If the file being transferred to the secure FTP server already exists at the destination, this property specifies whether the data from the file being transferred should be appended to the existing file. If set to **True**, the data is appended. If set to **False**, the file at the destination server is overwritten.<br /><br /> **Default value:** False|
    | Folder Path | Specifies the folder path on the secure FTP server where the file is copied.|
    | Port | Specifies the port address for the secure FTP server on which the file transfer takes place.|
    | Server Address | Specifies the server name or IP address of the secure FTP server.|
    | Target File Name | Specifies the name with which the file is transferred to the secure FTP server. You can also use macros for the target file name.|

4.  Click **OK** and **OK** again to save settings.

## Use a newer WinSCP version

> To use a newer version of WinSCP with BizTalk Server, add an assembly redirection so BizTalk knows which assembly to load. The redirection is configured in the BizTalk Server configuration files: BTSNTSVC.exe.config (32-bit host instances) and BTSNTSVC64.exe.config (64-bit host instances). Due to a breaking change in WinSCP automation after version 6.1.2, BizTalk 2016 (all versions) and BizTalk 2020 before CU6 can only be redirected up to maximum 6.1.2 version. BizTalk 2020 CU6 and later can be redirected to newer versions. 

Reviewer: Revised for easier scanning and parsing:

```suggestion
To use a newer version of WinSCP with BizTalk Server, add an assembly redirection so that BizTalk knows the correct assembly to load. You can configure this redirection in the following BizTalk Server configuration files: 

- BTSNTSVC.exe.config (32-bit host instances)
- BTSNTSVC64.exe.config (64-bit host instances)

You can redirect BizTalk 2020 CU6 and later to newer versions. However, due to a breaking change in WinSCP automation version 6.1.2 and later, you can only redirect BizTalk 2016 (all versions) and BizTalk 2020 before CU6 up to the 6.1.2 version at the latest.
```

Author: I change to this which is more correct as it is after 6.1.2, not 6.1.2 itself : 
To use a newer version of WinSCP with BizTalk Server, add an assembly redirection so that BizTalk knows the correct assembly to load. You can configure this redirection in the following BizTalk Server configuration files: 
- BTSNTSVC.exe.config (32-bit host instances)
- BTSNTSVC64.exe.config (64-bit host instances)
You can redirect BizTalk 2020 CU6 and later to newer versions. However, due to a breaking change in WinSCP automation **after** version 6.1.2, you can only redirect BizTalk 2016 (all versions) and BizTalk 2020 before CU6 up to the 6.1.2 version at the latest.

> The following includes sample configuration syntax. Be sure to replace `%NEWVERSION%` with your version of WinSCPnet.dll file:

Reviewer: ```suggestion
The following includes a sample configuration syntax. Make sure that you replace **`%NEWVERSION%`** with your version of the **WinSCPnet.dll** file:
```

| Is file mask case sensitive? | No. *.txt and \*.TXT works alike. Please install the latest cumulative update for BizTalk Server 2013. BizTalk Server 2013 RTM release had case-sensitive file masks. |
| Exporting bindings give a blank password field. When trying to create a receive location by importing these bindings what all changes are to be made? | Edit the binding file by editing the password field. Also, in `<Password vt="1">MySecretPassword</Password>`, **vt="1"** indicates a null value. Change that to **vt="8"**, which indicates a string. For example:<br /><br /> `<Password vt="8">MySecretPassword</Password>`<br /><br /> For more details, see [VarEnum Enum](/dotnet/api/system.runtime.interopservices.varenum) |
| How do I specify the file paths? | Normally, paths are specified in the format `/folder/pathname`. However, different servers require different formats, with or without leading or trailing slashes. So, you can also try the following:<br /><br /> -                   `/folder/pathname`<br /><br /> -                   `/folder/pathname/`<br /><br /> -                   `folder/pathname`<br /><br /> -                   `folder/pathname/` |

> | What is a good setting for Polling interval? | If you set this to a low value you may get high CPU consumption. It is recommended to set this to as high value as possible. You should also consider that only the DIR listing can take very long time if there are lots of files in the folder you poll from. |

Reviewer: ```suggestion
| What is a good setting for the polling interval? | If you set this interval to a low value, you might get high CPU consumption. The recommendation is that you set this interval to as high a value as possible. Also, take into consideration that if the folder where you're polling has many files, only the DIR listing can take a long time. |
```
